wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -3578,23 +3676,83 @@ WOLFSSL_LOCAL int tsip_SignRsaPkcs(wc_CryptoInfo* info, TsipUserCtx* tuc)
         else if (ssl->options.hashAlgo == sha256_mac)
             tsip_hash_type = R_TSIP_RSA_HASH_SHA256;
         else
-            ret = CRYPTOCB_UNAVAILABLE; 
+            ret = CRYPTOCB_UNAVAILABLE;
     }
-
+#else
+    (void)ssl;
+    
     if (ret == 0) {
+       if (tuc->sing_hash_type == md5_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_MD5;
+       else if (tuc->sing_hash_type == sha_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA1;
+       else if (tuc->sing_hash_type == sha256_mac)
+           tsip_hash_type = R_TSIP_RSA_HASH_SHA256;
+       else
+           ret = CRYPTOCB_UNAVAILABLE;
+    }
+    
+    switch (tuc->wrappedKeyType) {
+        case TSIP_KEY_TYPE_RSA1024:
+            if (tuc->keyflgs_crypt.bits.rsapri1024_key_set != 1)
+            {
+                WOLFSSL_MSG("tsip rsa private key 2048 not set");
+                    ret = CRYPTOCB_UNAVAILABLE;
+            }
+            break;
+        case TSIP_KEY_TYPE_RSA2048:
+            if (tuc->keyflgs_crypt.bits.rsapri2048_key_set != 1)
+            {
+                WOLFSSL_MSG("tsip rsa private key 2048 not set");
+                    ret = CRYPTOCB_UNAVAILABLE;
+            }
+            break;
+        default:
+            WOLFSSL_MSG("wrapped private key is not supported");
+            ret = CRYPTOCB_UNAVAILABLE;
+            break;
+    }
+    
+#endif
+    
+    if (ret == 0) {
+       #ifndef WOLFSSL_RENESAS_TSIP_CRYPTONLY
         hashData.pdata      = (uint8_t*)ssl->buffers.digest.buffer;
         hashData.data_type  = 1;
         sigData.pdata       = (uint8_t*)info->pk.rsa.in;
         sigData.data_length = 0; /* signature size will be returned here */
-
+       #else
+        hashData.pdata      = (uint8_t*)info->pk.rsa.in;
+        hashData.data_length= info->pk.rsa.inLen;
+        hashData.data_type  = tuc->keyflgs_crypt.bits.message_type;
+        sigData.pdata       = (uint8_t*)info->pk.rsa.out;
+        sigData.data_length = 0;
+       #endif
         if ((ret = tsip_hw_lock()) == 0) {
             switch (tuc->wrappedKeyType) {
-                
+            #ifdef WOLFSSL_RENESAS_TSIP_CRYPTONLY
+                case TSIP_KEY_TYPE_RSA1024:
+                    err = R_TSIP_RsassaPkcs1024SignatureGenerate(
+                                                &hashData, &sigData,
+                                       (tsip_rsa1024_private_key_index_t*)
+                                                tuc->rsa1024pri_keyIdx,
+                                                tsip_hash_type);
+
+                    if (err != TSIP_SUCCESS) {
+                        ret = WC_HW_E; 
+                    }
+                    break;
+            #endif
                 case TSIP_KEY_TYPE_RSA2048:
 
                     err = R_TSIP_RsassaPkcs2048SignatureGenerate(
                                                 &hashData, &sigData,
+                                   #ifndef WOLFSSL_RENESAS_TSIP_CRYPTONLY
```

Comment:
“#ifdef WOLFSSL_RENESAS_TSIP_TLS” seems more correct here. Does it fit your intentions?

Reply:
Thanks, Fixed